src/spikeinterface/postprocessing/amplitude_scalings.py

```diff
@@ -0,0 +1,304 @@
+        )
+        processor = ChunkRecordingExecutor(
+            recording,
+            func,
+            init_func,
+            init_args,
+            handle_returns=True,
+            job_name="extract amplitude scalings",
+            **job_kwargs,
+        )
+        out = processor.run()
+        (amp_scalings,) = zip(*out)
+        amp_scalings = np.concatenate(amp_scalings)
+
+        self._extension_data[f"amplitude_scalings"] = amp_scalings
+
+    def get_data(self, outputs="concatenated"):
+        """
+        Get computed spike amplitudes.
+        Parameters
+        ----------
+        outputs : str, optional
+            'concatenated' or 'by_unit', by default 'concatenated'
+        Returns
+        -------
+        spike_amplitudes : np.array or dict
+            The spike amplitudes as an array (outputs='concatenated') or
+            as a dict with units as key and spike amplitudes as values.
+        """
+        we = self.waveform_extractor
+        sorting = we.sorting
+
+        if outputs == "concatenated":
+            return self._extension_data[f"amplitude_scalings"]
+        elif outputs == "by_unit":
+            amplitudes_by_unit = []
+            for segment_index in range(we.get_num_segments()):
+                amplitudes_by_unit.append({})
+                segment_mask = self.spikes["segment_ind"] == segment_index
+                spikes_segment = self.spikes[segment_mask]
+                amp_scalings_segment = self._extension_data[f"amplitude_scalings"][segment_mask]
+                for unit_index, unit_id in enumerate(sorting.unit_ids):
+                    unit_mask = spikes_segment["unit_ind"] == unit_index
+                    amp_scalings = amp_scalings_segment[unit_mask]
+                    amplitudes_by_unit[segment_index][unit_id] = amp_scalings
+            return amplitudes_by_unit
+
+    @staticmethod
+    def get_extension_function():
+        return compute_amplitude_scalings
+
+
+WaveformExtractor.register_extension(AmplitudeScalingsCalculator)
+
+
+def compute_amplitude_scalings(
+    waveform_extractor,
+    sparsity=None,
+    max_dense_channels=16,
+    ms_before=None,
+    ms_after=None,
+    load_if_exists=False,
+    outputs="concatenated",
+    **job_kwargs,
+):
+    """
+    Computes the amplitude scalings from a WaveformExtractor.
+
+    Parameters
+    ----------
+    waveform_extractor: WaveformExtractor
+        The waveform extractor object
+    sparsity: ChannelSparsity
+        If waveforms are not sparse, sparsity is required if the number of channels is greater than
+        `max_dense_channels`. If the waveform extractor is sparse, its sparsity is automatically used.
+        By default None
+    max_dense_channels: int, optional
+        Maximum number of channels to allow running without sparsity. To compute amplitude scaling using
+        dense waveforms, set this to None, sparsity to None, and pass dense waveforms as input.
+        By default 16
```

Comment:
```suggestion
    max_dense_channels: int, default: 16
        Maximum number of channels to allow running without sparsity. To compute amplitude scaling using
        dense waveforms, set this to None, sparsity to None, and pass dense waveforms as input.
```

this is strict Numpy Docstring format

Reply:
I see, note that VS code automatically gives the `optional` notation when it generates docstrings

Reply:
That's the reason that this usually escpes me as well.